tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -143,6 +144,118 @@ def test_fail_on_multiple_drivers():
         extract_db_type_from_uri("mysql+pymsql+pyodbc://...")
 
 
+@pytest.fixture
+def test_database(tmp_path: Path) -> SqlAlchemyStore:
+    store = _get_store(tmp_path)
+    yield store
+    _cleanup_database(store)
+
+
+def _get_store(tmp_path: Path):
+    db_uri = f"{DB_URI}{tmp_path / 'temp.db'}"
+    MLFLOW_TRACKING_URI.set(db_uri)
+    return SqlAlchemyStore(db_uri, str(tmp_path))
```

Comment:
```suggestion
    db_uri = uri if uri := MLFLOW_TRACKING_URI.get() eles f"{DB_URI}{tmp_path / 'temp.db'}"
    return SqlAlchemyStore(db_uri, str(tmp_path))
```

Reply:
Would you mind elaborating on the intentions? I think it's confusing since we don't create a new store inside the `tmp_path` when the environment var for MLFLOW_TRACKING_URI is set. It might prevent us from isolation each test case.

Reply:
The original code has `MLFLOW_TRACKING_URI.get()`. This is for running this test suite against different databases such as PostgresSQL using an environment variable.

Reply:
So is my following understanding correct?

- This code is intended to enable us to conduct tests for other data stores like Postgres, but that is not included in the ci pipeline
- We want to create/delete the database storage file itself when the test is executed for sqlite, but want to reuse the DDL and just delete all of the table content for the other type of database

Reply:
Actually, I tried a similar code and saw the following error
```
E           mlflow.exceptions.MlflowException: Detected out-of-date database schema (found version 2d6e25af4d3e, but expected acf3f17fdcc7). Take a backup of your database, then run 'mlflow db upgrade <database_uri>' to migrate your database to the latest schema. NOTE: schema migration may result in database downtime - please consult your database's documentation for more detail.

current_rev = '2d6e25af4d3e'
engine     = Engine(sqlite:////workspaces/mlflow/mlruns.db)
head_revision = 'acf3f17fdcc7'
```

Reply:
I've addressed your comment in https://github.com/mlflow/mlflow/pull/10103/commits/c85b2b84494cd695d327bf751dd73c7d3ff06556, please take a look when you have time.

Reply:
> We want to create/delete the sqlite database storage file itself when MLFLOW_TRACKING_URI is not set for each test function, but want to reuse the DDL and just delete all of the table content for the other type of database in tear-down

Correct. 

> Do we want to separate the logic of sqlite db initialization so that the intention becomes obvious and the tear-down logic becomes consistent?

Sounds good. If it's not overcomplex. A follow-up PR is also fine.

Reply:
Sure, please let me keep this PR as it is and do the refactor later.

Reply:
let's do the refactor later